---
title: "SVMDO-Tutorial"
author: "<h4>Author: <i>Mustafa Erhan Özer</i></h4>" 
date: "<h4>Most recent update: <i>`r format( Sys.Date(), '%b-%d-%Y')`</i></h4>"

output:
    BiocStyle::html_document:
      toc: true
bibliography: citations.bib
link-citations: yes
csl: bioinformatics.csl

vignette: >
  %\VignetteIndexEntry{SVMDO-Tutorial}
  %\VignetteEngine{knitr::rmarkdown}
  %\VignetteEncoding{UTF-8}
---

```{r setup, include = FALSE}
knitr::opts_chunk$set(
  collapse = TRUE,
  warning = FALSE,
  comment = "#>",
  cache = TRUE
)


is_windows <- identical(.Platform$OS.type, "windows")
```
# Installation

The day-to-day development version from the [Github repository](https://github.com/robogeno/SVMDO) can be installed.
```{r install, eval=FALSE}
# From Github 
if (!requireNamespace("devtools", quietly=TRUE))
  install.packages("devtools")
devtools::install_github("robogeno/SVMDO")
```

```{r eval = FALSE}
# From Biocodunctor
if(!requireNamespace("BiocManager", quietly = TRUE))
    install.packages("BiocManager")
BiocManager::install("SVMDO")
```


# Package Loading
```{r load_windows, message=FALSE, warning=FALSE, eval=is_windows, echo=is_windows}
#library(SVMDO)
#Main screen of GUI
#runGUI()
# OR
#SVMDO::runGUI()
```

```{r load_other_os, message=FALSE, warning=FALSE, eval=isFALSE(is_windows), echo=isFALSE(is_windows)}
#library(SVMDO)
#runGUI()
# OR
#SVMDO::runGUI()
```

# Introduction
Transcriptome-based supervised classification can be difficult to handle due to including vast amount of junk features reducing the efficiency of the process. To minimize the efficiency problem, feature selection methods have been applied [@Cai2018]. Ensemble methodology in constructing a feature selection method has been mainly preferred for its better accuracy in classification processes [@Liu2004]. Cancer-based studies have been reported to co-occur with several chronic diseases [@Gallagher2015] and oncogenic viral infections [@Kori2020]. Disease Ontology (DO) enrichment analysis can provide a filtration approach through reported diseases that interact with the gene sets used [@LePendu2011]. Wilk's lambda criterion is one of the moderate techniques for feature selection. The method combines different features(genes) step by step based on their contributions to the discriminating power of overall model [@elOuardighi2007]. Thus, disease related genes with better discriminatory characteristics can be acquired. SVMDO is an easy-to-use GUI using disease information for detecting tumor/normal sample discriminating gene sets from differentially expressed genes. Our approach is based on an iterative algorithm filtering genes with disease ontology enrichment analysis and wilk’s lambda criterion connected to SVM classification model construction. Along with gene set extraction, SVMDO also provides individual prognostic marker detection. The algorithm is designed for FPKM and RPKM normalized RNA-Seq transcriptome datasets. During the development our algorithm, Bioconductor provided a robust approach for acquiring significant disease-gene interaction information from disease ontology database.


# Implementation
SVMDO was developed by using Shiny package. It is available for Windows and Linux from Bioconductor website. SVMDO requires the following R packages: Shiny, ShinyFiles, golem, nortest, e1071, BSDA, data.table, sjmisc, klaR, caTools, dplyr, caret, survival, DOSE, AnnotationDbi, DOSE, org.Hs.eg.db

# Dataset Preparation
RNA-Seq cancer transcriptome and clinical datasets must be prepared before applying them in SVMDO. The datasets must be used in **txt** format. The expected designs of input datasets are indicated in Table-1 and Table-2 for transcriptome and clinical datasets respectively.

```{r table_data_prep_1, echo=FALSE}
id<-c("TCGA-AA-3662","TCGA-AA-3514","TCGA-D5-6541","...")
tissue_type<-c("Normal","Normal","Tumour","...")
AB1G<-c("Exp_1","Exp_2","Exp_3","...")
A2M<-c("Exp_1","Exp_2","Exp_3","...")
table_prep_1<-data.frame(id,tissue_type,AB1G,A2M)
```
`r knitr::kable(table_prep_1,"pipe",col.name=c("id","tissue_type","AB1G","A2M"),align = "cccc",caption="Table-1 Structure of RNA-Seq Transcriptome Dataset")`

```{r table_data_prep_2, echo=FALSE}
id<-c("TCGA-AA-3662","TCGA-AA-3514","TCGA-D5-6541","...")
days_to_death<-c(49,1331,225,"...")
vital_status<-c("Alive","Dead","Dead","...")
table_prep_2<-data.frame(id,days_to_death,vital_status)
```
`r knitr::kable(table_prep_2,"pipe",col.name=c("id","days_to_death","vital_status"),align = "cccc",caption="Table-2 Structure of Clinical Dataset")`

* **tissue_type:** Normal/Tumour (or Tumor) Tissue Information 
* **A1BG, A2M:** Gene Symbol and Gene Expression
* **id:** TCGA Sample Ids
* **days_to_death:** Survival Time of Patients 
* **vital_status:** Vitality of Patients

Using **tissue_type** and **id** as column names for representing tissue information and TCGA sample id is **mandatory**. If there is not any requirement for **survival analysis**, preparation of **clinical dataset** and involvement of **id** column in the transcriptome dataset are optional.

# SVMDO GUI Description
The main dialog box is indicated in **Figure-1**. The GUI page consists of two main sections which are "Analysis" and "Result". In the Analysis section, steps of acquiring discriminative gene set and preparations of individual survival plots are included. In the Result section, visualization and download of discriminative gene set and survival plots are included. At each step of the GUI sections, necessary variables are saved as objects in the workspace environment to be used in the following sections. The input of SVMDO is a txt file of transcriptome dataset prepared in the form of **Table-1**. 

```{r, out.width = "30%",fig.align='center',fig.show='hold',echo=FALSE}
# two figs side by side
knitr::include_graphics(c("svmdo_analysis_screen.png",
                   "svmdo_result_screen.png"))
```

# How to open SVMDO Main Screen
To open GUI screen, you can directly write **SVMDO::runGUI()** in R console. If library is previously activated you can open GUI screen by writing **runGUI()** in R console.

# How to use SVMDO Main Screen

Steps of Analysis Screen

1. To change your output directory, you can adjust it in the **Choose Directory** section.  Otherwise, default output directory will automatically be used.
2. To search your transcriptome dataset, use the file detection in **Choose Your Expression Dataset** section. The file will be automatically uploaded into the GUI.
3. To prevent clashing with test datasets, select **"None"** option from the radio button section.
4. By clicking on **DEG Analysis** button you further apply differential expression analysis. Labels of tissue_type column in dataset must contain “Nor” and “Tum” for determining normal/tumour (or tumor) samples. A message window saying **Process Completed** will appear if there is not any problem.
5. When the differential expression process is completed, a user-defined input size (n) is selected to filter the initial gene list (i.e., n number of upregulated and downregulated genes) by entering a number in **Input Size** section. It is predetermined as **50** in GUI which can be changed based on the user. A meesage window saying **process completed** will appear if there is not any problem. If there is problem with the value of input size, you will get a warning about inappropriate input size selection. If the input size remains, algorithm selects all of the differentiallly expressed genes to be used in the next process.
6. To apply disease ontology-based gene filtration, click on **DO Analysis** button. A message window saying **process completed** will appear if there is not any problem.
7. To further apply the following feature selection and classification processes, click on the **Classification** button. A message window saying **process completed** will appear if there is not any problem.
8. Acquired discriminative gene set can be further used for survival analysis to detect individual prognostic genes. To apply this process, use the file detection in **Choose Clinical Data** section for searching clinical data about patient survival followed by clicking on **Survival Analysis** button. A meesage window saying **process completed** will appear if there is not any problem.

Applications of Result Screen

1. To visualize discriminative gene sets inside GUI screen, click on **Show Gene Results** button. When you click this button, a table of gene set will appear. The resulting table can be downloaded by clicking on **Download Gene List** button.If there is a problem in the analysis, an error message will appear.
2. To visualize survival plots of individual genes, two steps have to be applied. First of all, click on **Prepare Plot Lists** button to feed plot information to the visualization system. After that, click on **Show Plots** button to visualize survival plots. Once **Prepare Plot Lists** button is clicked, the resulting plot list can be downloaded by clicking on **Download Plot List** button. If there is a problem in the analysis, an error message will appear at each step.

# Using test datasets with SVMDO Main Screen
Before using the GUI in a study, analysis using the test datasets can be used to understand overall functionality of the algorithm. Test datasets consists of objects of expression and Clinical datasets integrated into summarized experiment objects. These objects are saved in RDA forms and called from **extdata** folder of the package. 

When test datasets are used, predefined input size (n=50) is used. 
1. First of all, select the dataset to be used in test analysis by using radio buttons. There are two datasets consisting of simplified forms of TCGA-COAD (COAD) and TCGA-LUSC (LUSC) with 400 genes loaded into summarized experiement objects.
2. To apply differential expression analysis on the selected dataset click on the **DEG Analysis** button. A meesage window saying **process completed** will appear if there is not any problem.
3. To further apply disease ontology-based gene filtration, click on **DO Analysis** button. A meesage window saying **process completed** will appear if there is not any problem.
4. To apply survival analysis on the acquired discriminative gene sets, click on **Survival Analysis** button. Clinical dataset for the selected dataset is automatically loaded into the GUI. A meesage window saying **process completed** will appear if there is not any problem.

# Workspace Clearance
When the user task is completed, click on the **Clear Environment** button to remove the global variables created during the algorithm sections. To prevent error in the next usages of GUI, it is a necessary process. It can be applied at any moment without the necessity of completing all of the steps of algorithm. A message window saying **process completed** will appear if there is not any problem.

# Output Files of SVMDO
* **final_discriminative_gene_set:** Final form of gene set providing optimal classification performance 
* **Individiual Survival Plots:** Separate survival plots of individual genes with prognostic performance

# References
<div id="refs"></div>

# How to install R and RStudio GUI

## Windows Operating System

**R-base Install:

1. Copy/Paste the address https://cran.r-project.org/bin/windows/base/
2. Click on Download R-X.X.X for Windows
3. Click on the downloaded R-X.X.X-win.exe file to install R-base

**RStudio Install:

1. Copy/Paste the address  https://posit.co/products/open-source/rstudio/
2. In RStudio Desktop Section, select open-source edition
3. In All Installers section, select .exe or .zip version for Windows to download RStudio-X.X.X.exe or RStudio-X.X.X.zip file
4. Click on the downloaded RStudio-X.X.X.exe and/or unzipped file to install RStudio

## Ubuntu Operating System

**R-base Install:

1. Copy/Paste the address https://cran.r-project.org/
2. In the Download and Install R section, click on Download R for Linux
3. In the parent directory, click on ubuntu/ 
4. Run the following lines in terminal (as root or by prefixing sudo) to provide access to recent version of R:
* apt update -qq
* apt install --no-install-recommends software-properties-common dirmngr
* wget -qO- https://cloud.r-project.org/bin/linux/ubuntu/marutter_pubkey.asc | sudo tee -a /etc/apt/trusted.gpg.d/cran_ubuntu_key.asc
* add-apt-repository "deb https://cloud.r-project.org/bin/linux/ubuntu $(lsb_release -cs)-cran40/"

5. Run apt-get install r-base

**RStudio Install:

1. Copy/Paste the address https://posit.co/products/open-source/rstudio/
2. In RStudio Desktop Section, select open-source edition
3. In All Installers section, select .deb or .debian.tar.gz version for Ubuntu 
4. Right click on the download and/or unzipped .deb file to install RStudio
   (Alternatively, .deb file can be installed from terminal by typing 
   sudo dpkg -i installed_Rstudio_file.deb)


# Session Info
```{r SessionInfo, eval=TRUE, message=FALSE, echo=FALSE}
sessionInfo()
```

